tests/profiles-enabled/test_shipments.py

```diff
@@ -1800,3 +1803,204 @@ def test_shipment_set_shadow(self):
+
+            # ------------------------- Results Pagination test -----------------------#
+            mock_get.reset_mock()
+            # The api call returns 15 results which should yield 2 pages.
+            self.map_responses = {self.iot_enpoints[0]: self.iot_responses(OWNER_ID, num_devices=15)}
+            response = self.client.get(url)
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+            page = response.json()
+            # Two pages results
+            self.assertEqual(len(page['data']), 10)
+            self.assertTrue(page.get('meta', None))
+            self.assertTrue(page.get('links', None))
+            self.assertEqual(page['meta']['pagination']['pages'], 2)
+            self.assertEqual(page['meta']['pagination']['count'], 15)
+            self.assertTrue(page['links']['next'])
+
+            # ---------------------------- in_bbox param validation test ----------------------------#
+            mock_get.reset_mock()
+            # A call with a non numeric value in in_bbox should fail with 400 status
+            in_bbox_url = f'{url}?in_bbox=-82.5,34.5,-82a,35'
+            response = self.client.get(in_bbox_url)
+            self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+            # A call with a longitude value out of range in in_bbox should fail with 400 status
+            in_bbox_url = f'{url}?in_bbox=-181,34.5,-82,35'
+            response = self.client.get(in_bbox_url)
+            self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+            # A call with a latitude value out of range in in_bbox should fail with 400 status
+            in_bbox_url = f'{url}?in_bbox=-82.5,34.5,-82,95'
+            response = self.client.get(in_bbox_url)
+            self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+            # A call with less than 4 in_bbox parameters should fail with 400 status
+            in_bbox_url = f'{url}?in_bbox=-82.5,34.5,-82'
+            response = self.client.get(in_bbox_url)
+            self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+            # A call with more than 4 in_bbox parameters should fail with 400 status
+            in_bbox_url = f'{url}?in_bbox=-82.5,34.5,-82,45,-25'
+            response = self.client.get(in_bbox_url)
+            self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+            # A call with well formed in_bbox parameters should succeed with 200 status
+            # even with a couple of blank spaces between values
+            in_bbox_url = f'{url}?in_bbox=-82.5 ,34.5,-82, 35'
+            self.map_responses = {self.iot_enpoints[4]: self.iot_responses(OWNER_ID)}
+            response = self.client.get(in_bbox_url)
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+            # valid box should respects the rule: the third element should be greater than the first
```

Comment:
[A] valid [b]box should respect[] the rule: the third element should be greater than the first